### PR TITLE
chore: update tooling

### PR DESCRIPTION
**Pull request by raisedadead**

### CODE REVIEW

**Comment by raisedadead** [comment]
I have integrated most of the tooling and chore stuff. 

- With this when you do a `npm run commit` the code will be lint-ed and automatically formatted.
- You will be able to have nice standard commit messages.
- We can expand the scopes as need arises.

Bonus, added all of the snyk stuff to CI.

**Comment by raisedadead** [comment]
Yup, again the snyk error is expected coming in from a devDependency.

**Comment by raisedadead** [comment]
Note: I have knowingly NOT used our own adapter (cz-freecodecamp) here, its quite obsolete. And needs refactoring in its own code to be generic.

Thats's fine which we can ignore for this project and get back to later.

File: commitizen.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    name: "test:     Adding missing tests"
+  },
+  {
+    value: "chore",
+    name: `chore:    Changes to the build process or auxiliary tools
+            and libraries such as documentation generation`
+  },
+  {
+    value: "revert",
+    name: "revert:   Revert to a commit"
```
Comment (ojongerius) [inline commitizen.config.js]: Maybe "revert a commit" -as you revert commits, and record new commits for them.

**Review by ojongerius — CHANGES_REQUESTED** [review]

**Pull request merged**